crates/pop-contracts/src/init_tests.rs

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: GPL-3.0
+
+use crate::{create_smart_contract, Contract};
+use anyhow::Result;
+use std::{fs, path::PathBuf};
+
+pub fn generate_smart_contract_test_environment() -> Result<tempfile::TempDir> {
+	let temp_dir = tempfile::tempdir().expect("Could not create temp dir");
+	let temp_contract_dir = temp_dir.path().join("testing");
+	fs::create_dir(&temp_contract_dir)?;
+	create_smart_contract("testing", temp_contract_dir.as_path(), &Contract::Standard)?;
+	Ok(temp_dir)
+}
+
+// Function that mocks the build process generating the contract artifacts.
```

Comment:
Doc comment for public function. Please see all clippy warnings in this PR of the same nature.